cpp/src/fil/common.cuh

```diff
@@ -143,14 +157,16 @@ struct shmem_size_params {
   int block_dim_x = 0;
   /// shm_sz is the associated shared memory footprint
   int shm_sz = INT_MAX;
+  /// sizeof_real is the size in bytes of all floating-point variables during inference
+  std::size_t sizeof_real = 4;
```

Comment:
Is this a fixed constant? In that case, I would suggest this change so the compiler substitutes this value at compile time
```suggestion
  constexpr std::size_t sizeof_real = 4;
```

Reply:
The plan is for to be a variable with the value of either 4 or 8 (i.e. `sizeof(float)` or `sizeof(double)`). Making it a constant would require templatizing `shmem_size_params`, which I think would be more complex.